src/app/add-data-set-dialog/add-data-set-dialog.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from '@angular/core';
+
+import { HttpHeaders } from '@angular/common/http';
+
+import { RequestService } from '../request/request.service'
+import { MatDialogRef} from "@angular/material/dialog";
+
+interface Project {
+  name: string;
+  url: string;
+}
+
+@Component({
+  selector: 'app-add-data-set-dialog',
+  templateUrl: './add-data-set-dialog.component.html',
+  styleUrls: ['./add-data-set-dialog.component.css']
+})
+
+export class AddDataSetDialogComponent implements OnInit {
+
+  projects: Project[] = [{name: '', url: ''}];
```

Comment:
Is "dataset" a better name than "projects"?

@simsimkic While on the subject, do we need to introduce the concept of a "DatasetProject" on the backend of things? The dataset contains projects, which contain instances. The dataset can be defined by a name (e.g., Clean CaDET, MLCQ...) while the projects have a name and url. What do you think?